enarxbot.py

```diff
@@ -58,3 +59,31 @@ def create_card(column, content_id, content_type):
         if error["resource"] != "ProjectCard" or error["code"] != "unprocessable":
             raise
         print("Card already in project.")
+
+# Takes a project as input and caches all cards in a dictionary for easy lookup
+# (and to avoid redundant network requests).
+def cache_board(project):
+    card_cache = {}
+    for column in project.get_columns():
+            for card in column.get_cards():
+                content = card.get_content()
+                if content:
+                    card_cache[content.id] = (card, column)
+    return card_cache
+
+# Get all issue numbers related to a PR.
+def get_related_issues(pr):
+    # Regex to pick out closing keywords.
+    regex = re.compile("(close[sd]?|fix|fixe[sd]?|resolve[sd]?)\s*:?\s+#(\d+)", re.I)
+    
+    related_issues = set()
+
+    # Extract all associated issues from closing keyword in PR
+    for verb, num in regex.findall(pr.body):
+        related_issues.add(int(num))
+
+    # Extract all associated issues from PR commit messages
+    for c in pr.get_commits():
+        for verb, num in regex.findall(c.commit.message):
+            related_issues.add(int(num))
+    return related_issues
```

Comment:
Use `yield` in this function instead of a set.

Reply:
@npmccallum I was using `yield` before, but that included duplicates (which caused bugs downstream). Maybe it's not a problem anymore -- I've changed it back to `yield`.